### PR TITLE
image on scroll

### PR DESCRIPTION
**Pull request by jaydgoss**
node_modules/ember-in-viewport/addon/utils/is-in-viewport.js  needs to be altered for elements larger than the viewport to work.
https://github.com/michaeldeol/ember-in-viewport/commit/d4e182a8380a41ea8f71d2394d67aff27af3eda0

### CODE REVIEW

**Review by sepowitz — APPROVED** [review]

File: app/components/scroll-image.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+} = Ember;
+
+export default Component.extend(InViewportMixin, {
+  v: v(c),
+  classNames: ['absolute', 't0'],
+  sanctu: service(),
+  active: false,
+  random: null,
+  max: $(window).width() - 200,
+  tolerance: $(window).height() * -0.25,
```
Comment (hhff) [inline app/components/scroll-image.js]: These won't recalc on resize - is that intentional?

Comment (jaydgoss) [reply app/components/scroll-image.js]: Not intentional

File: app/styles/app.css [hunk 2]
```diff
@@ -13,10 +13,38 @@ body > .ember-view {
...
+  top: 0px;
+}
+
+.opac1 {
+  opacity: 1;
+}
+
+.opac0 {
+  opacity: 0;
+}
```
Comment (hhff) [inline app/styles/app.css]: we can lose this

File: app/components/scroll-image.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+const {
+  Component,
+  inject: { service },
+  $,
+} = Ember;
+
+export default Component.extend(InViewportMixin, {
+  v: v(c),
+  classNames: ['absolute', 't0'],
```
Comment (hhff) [inline app/components/scroll-image.js]: these can be vudu-ified by:

```

const v = v(c)
export default Component.extend(InViewportMixin, {
  v: v,
  classNames: [v.absolute, v.t0],
```

File: app/components/scroll-image.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+   this.random = `${this.randomN}px`,
+   this.handleResize = () => {
+     let nW = $(window).width();
+     let nH = $(window).height();
+     let wDif = nW - this.iW;
+     if (this.iW !== nW) this.set('random', `${this.randomN + wDif}px`);
+     if (this.iH !== nH) this.set('tolerance', nH * -0.25);
+    };
+    Ember.run.next(this, this.handleResize);
+    $(window).on('resize', Ember.run.bind(this, this.handleResize));
```
Comment (hhff) [inline app/components/scroll-image.js]:
```
   this.handleResize = Ember.run.bind(this, () => {
     let nW = $(window).width();
     let nH = $(window).height();
     let wDif = nW - this.iW;
     if (this.iW !== nW) this.set('random', `${this.randomN + wDif}px`);
     if (this.iH !== nH) this.set('tolerance', nH * -0.25);
   });
   Ember.run.next(this, this.handleResize);
   $(window).on('resize', this.handleResize);
```

then below,

```
willDestroyElement() {
  $(window).off('resize', this.handleResize);
}
```

Comment (hhff) [reply app/components/scroll-image.js]: reason is that jQuery uses the "place in memory" to find the function to remove

File: app/templates/index.hbs [hunk 5]
```diff
@@ -1,12 +1,21 @@
...
-        {{{section.copy}}}
-      {{/way-point}}
+    {{#each model as |group|}}
+      {{#scroll-section
+        class=(concat 'index--' group.title)
+        action=(action "didHitWaypoint" group.title null)
+      }}
+        <p class={{vudu 'pb5'}}>
+          {{#each group.combined as |c|}}
+            <span class={{if c.break "relative flex" "relative"}}>
```
Comment (hhff) [inline app/templates/index.hbs]: the way around this w/o classic CSS is:

```
<span class={{if c.break (concat v.relative " " v.flex) v.relative}}>
```

or you can declare classes in the `route/index.js` like:

```
let local = v({
  scrollSectionBreak: {
    "@composes: [v.relative, v.flex]
  }
})
```


<span class={{if c.break local.scrollSectionBreak v.relative}}>
```

**Pull request merged**